diff --git a/drivers/staging/rtl8712/rtl871x_xmit.c b/drivers/staging/rtl8712/rtl871x_xmit.c
index 8bbdee70..e2266e90 100644
--- a/drivers/staging/rtl8712/rtl871x_xmit.c
+++ b/drivers/staging/rtl8712/rtl871x_xmit.c
@@ -208,13 +208,13 @@ sint r8712_update_attrib(struct _adapter *padapter, _pkt *pkt,
 	pattrib->ether_type = ntohs(etherhdr.h_proto);
 
 {
-	u8 bool;
+	u8 Bool;
 	/*If driver xmit ARP packet, driver can set ps mode to initial
 	 * setting. It stands for getting DHCP or fix IP.*/
 	if (pattrib->ether_type == 0x0806) {
 		if (padapter->pwrctrlpriv.pwr_mode !=
 		    padapter->registrypriv.power_mgnt) {
-			_cancel_timer(&(pmlmepriv->dhcp_timer), &bool);
+			_cancel_timer(&(pmlmepriv->dhcp_timer), &Bool);
 			r8712_set_ps_mode(padapter, padapter->registrypriv.
 				power_mgnt, padapter->registrypriv.smart_ps);
 		}
diff --git a/include/linux/capability.h b/include/linux/capability.h
index a63d13d8..c1a97d8f 100644
--- a/include/linux/capability.h
+++ b/include/linux/capability.h
@@ -14,6 +14,7 @@
 #define _LINUX_CAPABILITY_H
 
 #include <linux/types.h>
+#include <stdbool.h>
 
 struct task_struct;
 
diff --git a/include/linux/cpumask.h b/include/linux/cpumask.h
index 4f7a6323..a4578534 100644
--- a/include/linux/cpumask.h
+++ b/include/linux/cpumask.h
@@ -9,6 +9,7 @@
 #include <linux/kernel.h>
 #include <linux/threads.h>
 #include <linux/bitmap.h>
+#include <stdbool.h>
 
 typedef struct cpumask { DECLARE_BITMAP(bits, NR_CPUS); } cpumask_t;
 
diff --git a/include/linux/kernel.h b/include/linux/kernel.h
index e8b1597b..849a5240 100644
--- a/include/linux/kernel.h
+++ b/include/linux/kernel.h
@@ -21,6 +21,7 @@
 #include <linux/dynamic_debug.h>
 #include <asm/byteorder.h>
 #include <asm/bug.h>
+#include <stdbool.h>
 
 #define USHRT_MAX	((u16)(~0U))
 #define SHRT_MAX	((s16)(USHRT_MAX>>1))
diff --git a/include/linux/log2.h b/include/linux/log2.h
index fd7ff3d9..aaf6073c 100644
--- a/include/linux/log2.h
+++ b/include/linux/log2.h
@@ -14,6 +14,7 @@
 
 #include <linux/types.h>
 #include <linux/bitops.h>
+#include <stdbool.h>
 
 /*
  * deal with unrepresentable constant logarithms
diff --git a/include/linux/stddef.h b/include/linux/stddef.h
index 6a40c76b..7b3e42dc 100644
--- a/include/linux/stddef.h
+++ b/include/linux/stddef.h
@@ -12,11 +12,6 @@
 
 #ifdef __KERNEL__
 
-enum {
-	false	= 0,
-	true	= 1
-};
-
 #undef offsetof
 #ifdef __compiler_offsetof
 #define offsetof(TYPE,MEMBER) __compiler_offsetof(TYPE,MEMBER)
diff --git a/include/linux/string.h b/include/linux/string.h
index e033564f..0ec7b0d1 100644
--- a/include/linux/string.h
+++ b/include/linux/string.h
@@ -11,6 +11,7 @@
 #include <linux/types.h>	/* for size_t */
 #include <linux/stddef.h>	/* for NULL */
 #include <stdarg.h>
+#include <stdbool.h>
 
 extern char *strndup_user(const char __user *, long);
 extern void *memdup_user(const void __user *, size_t);
diff --git a/include/linux/types.h b/include/linux/types.h
index 57a97234..8ec0b656 100644
--- a/include/linux/types.h
+++ b/include/linux/types.h
@@ -34,8 +34,6 @@ typedef __kernel_timer_t	timer_t;
 typedef __kernel_clockid_t	clockid_t;
 typedef __kernel_mqd_t		mqd_t;
 
-typedef _Bool			bool;
-
 typedef __kernel_uid32_t	uid_t;
 typedef __kernel_gid32_t	gid_t;
 typedef __kernel_uid16_t        uid16_t;
diff --git a/kernel/panic.c b/kernel/panic.c
index b2659360..c9c67232 100644
--- a/kernel/panic.c
+++ b/kernel/panic.c
@@ -160,8 +160,8 @@ EXPORT_SYMBOL(panic);
 
 struct tnt {
 	u8	bit;
-	char	true;
-	char	false;
+	char	True;
+	char	False;
 };
 
 static const struct tnt tnts[] = {
@@ -211,7 +211,7 @@ const char *print_tainted(void)
 		for (i = 0; i < ARRAY_SIZE(tnts); i++) {
 			const struct tnt *t = &tnts[i];
 			*s++ = test_bit(t->bit, &tainted_mask) ?
-					t->true : t->false;
+					t->True : t->False;
 		}
 		*s = 0;
 	} else
