diff --git a/drivers/uio/uio.c b/drivers/uio/uio.c
index a783d533..16f419d6 100644
--- a/drivers/uio/uio.c
+++ b/drivers/uio/uio.c
@@ -632,7 +632,7 @@ static int uio_vma_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
 	offset = (vmf->pgoff - mi) << PAGE_SHIFT;
 
 	if (idev->info->mem[mi].memtype == UIO_MEM_LOGICAL)
-		page = virt_to_page(idev->info->mem[mi].addr + offset);
+		page = virt_to_page((void *)(unsigned long)idev->info->mem[mi].addr + offset);
 	else
 		page = vmalloc_to_page((void *)(unsigned long)idev->info->mem[mi].addr + offset);
 	get_page(page);
