diff --git a/include/linux/moduleparam.h b/include/linux/moduleparam.h
index 52666d90..dedebe24 100644
--- a/include/linux/moduleparam.h
+++ b/include/linux/moduleparam.h
@@ -391,9 +394,11 @@ extern void param_free_charp(void *arg);
 
 /* We used to allow int as well as bool.  We're taking that away! */
 extern const struct kernel_param_ops param_ops_bool;
+#define param_ops__Bool param_ops_bool
 extern int param_set_bool(const char *val, const struct kernel_param *kp);
 extern int param_get_bool(char *buffer, const struct kernel_param *kp);
 #define param_check_bool(name, p) __param_check(name, p, bool)
+#define param_check__Bool(name, p) __param_check(name, p, bool)
 
 extern const struct kernel_param_ops param_ops_bool_enable_only;
 extern int param_set_bool_enable_only(const char *val,
diff --git a/include/linux/stddef.h b/include/linux/stddef.h
index f4aec0e7..46712c38 100644
--- a/include/linux/stddef.h
+++ b/include/linux/stddef.h
@@ -6,11 +6,6 @@
 #undef NULL
 #define NULL ((void *)0)
 
-enum {
-	false	= 0,
-	true	= 1
-};
-
 #undef offsetof
 #ifdef __compiler_offsetof
 #define offsetof(TYPE, MEMBER)	__compiler_offsetof(TYPE, MEMBER)
diff --git a/include/linux/types.h b/include/linux/types.h
index 4d118ba1..87502de8 100644
--- a/include/linux/types.h
+++ b/include/linux/types.h
@@ -3,6 +3,7 @@
 
 #define __EXPORTED_HEADERS__
 #include <uapi/linux/types.h>
+#include <stdbool.h>
 
 #ifndef __ASSEMBLY__
 
@@ -26,8 +27,6 @@ typedef __kernel_timer_t	timer_t;
 typedef __kernel_clockid_t	clockid_t;
 typedef __kernel_mqd_t		mqd_t;
 
-typedef _Bool			bool;
-
 typedef __kernel_uid32_t	uid_t;
 typedef __kernel_gid32_t	gid_t;
 typedef __kernel_uid16_t        uid16_t;
