diff --git a/drivers/staging/rtl8712/rtl871x_xmit.c b/drivers/staging/rtl8712/rtl871x_xmit.c
index 78f570b5..30f2b81f 100644
--- a/drivers/staging/rtl8712/rtl871x_xmit.c
+++ b/drivers/staging/rtl8712/rtl871x_xmit.c
@@ -204,13 +204,13 @@ sint r8712_update_attrib(struct _adapter *padapter, _pkt *pkt,
 	pattrib->ether_type = ntohs(etherhdr.h_proto);
 
 {
-	u8 bool;
+	u8 Bool;
 	/*If driver xmit ARP packet, driver can set ps mode to initial
 	 * setting. It stands for getting DHCP or fix IP.*/
 	if (pattrib->ether_type == 0x0806) {
 		if (padapter->pwrctrlpriv.pwr_mode !=
 		    padapter->registrypriv.power_mgnt) {
-			_cancel_timer(&(pmlmepriv->dhcp_timer), &bool);
+			_cancel_timer(&(pmlmepriv->dhcp_timer), &Bool);
 			r8712_set_ps_mode(padapter, padapter->registrypriv.
 				power_mgnt, padapter->registrypriv.smart_ps);
 		}
diff --git a/include/linux/stddef.h b/include/linux/stddef.h
index f4aec0e7..46712c38 100644
--- a/include/linux/stddef.h
+++ b/include/linux/stddef.h
@@ -7,11 +7,6 @@
 #undef NULL
 #define NULL ((void *)0)
 
-enum {
-	false	= 0,
-	true	= 1
-};
-
 #undef offsetof
 #ifdef __compiler_offsetof
 #define offsetof(TYPE,MEMBER) __compiler_offsetof(TYPE,MEMBER)
diff --git a/include/linux/types.h b/include/linux/types.h
index 4d118ba1..87502de8 100644
--- a/include/linux/types.h
+++ b/include/linux/types.h
@@ -3,6 +3,7 @@
 
 #define __EXPORTED_HEADERS__
 #include <uapi/linux/types.h>
+#include <stdbool.h>
 
 #ifndef __ASSEMBLY__
 
@@ -26,8 +27,6 @@ typedef __kernel_timer_t	timer_t;
 typedef __kernel_clockid_t	clockid_t;
 typedef __kernel_mqd_t		mqd_t;
 
-typedef _Bool			bool;
-
 typedef __kernel_uid32_t	uid_t;
 typedef __kernel_gid32_t	gid_t;
 typedef __kernel_uid16_t        uid16_t;
diff --git a/kernel/panic.c b/kernel/panic.c
index 6d630037..b601fe44 100644
--- a/kernel/panic.c
+++ b/kernel/panic.c
@@ -192,8 +192,8 @@ EXPORT_SYMBOL(panic);
 
 struct tnt {
 	u8	bit;
-	char	true;
-	char	false;
+	char	True;
+	char	False;
 };
 
 static const struct tnt tnts[] = {
@@ -243,7 +243,7 @@ const char *print_tainted(void)
 		for (i = 0; i < ARRAY_SIZE(tnts); i++) {
 			const struct tnt *t = &tnts[i];
 			*s++ = test_bit(t->bit, &tainted_mask) ?
-					t->true : t->false;
+					t->True : t->False;
 		}
 		*s = 0;
 	} else
