diff --git a/arch/mips/include/asm/page.h b/arch/mips/include/asm/page.h
index 0e095a41..c4246945 100644
--- a/arch/mips/include/asm/page.h
+++ b/arch/mips/include/asm/page.h
@@ -241,12 +241,12 @@ extern int __virt_addr_valid(const volatile void *kaddr);
 })
 #define CAC_ADDR(addr)							\
 ({									\
-	unsigned long __addr = (addr);					\
+	unsigned long __addr = (unsigned long)(addr);					\
 	if (__addr < UNCAC_BASE_UPPER)					\
 		__addr = __addr - UNCAC_BASE + PAGE_OFFSET;		\
 	else								\
 		__addr = __addr - UNCAC_BASE_UPPER + CAC_BASE_UPPER;	\
-	__addr;								\
+	(void *)__addr;								\
 })
 #elif defined(CONFIG_BRCM_UPPER_768MB)
 
diff --git a/arch/mips/mm/dma-default.c b/arch/mips/mm/dma-default.c
index 541e490a..4365941e 100644
--- a/arch/mips/mm/dma-default.c
+++ b/arch/mips/mm/dma-default.c
@@ -158,7 +158,7 @@ static void mips_dma_free_coherent(struct device *dev, size_t size, void *vaddr,
 	addr = (unsigned long)brcm_unmap_coherent(vaddr);
 #else
 	if (!plat_device_is_coherent(dev))
-		addr = CAC_ADDR(addr);
+		addr = (long unsigned int)CAC_ADDR(addr);
 #endif
 
 	free_pages(addr, get_order(size));
diff --git a/kernel/extable.c b/kernel/extable.c
index fe35a634..5401b494 100644
--- a/kernel/extable.c
+++ b/kernel/extable.c
@@ -52,7 +52,7 @@ const struct exception_table_entry *search_exception_tables(unsigned long addr)
 {
 	const struct exception_table_entry *e;
 
-	e = search_extable(__start___ex_table, __stop___ex_table-1, addr);
+	e = search_extable(__start___ex_table, (const unsigned int)__stop___ex_table-1, addr);
 	if (!e)
 		e = search_module_extables(addr);
 	return e;
diff --git a/kernel/module.c b/kernel/module.c
index eab08274..1e8afc8d 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -3686,7 +3686,7 @@ const struct exception_table_entry *search_module_extables(unsigned long addr)
 			continue;
 
 		e = search_extable(mod->extable,
-				   mod->extable + mod->num_exentries,
+				   (const unsigned int)mod->extable + mod->num_exentries,
 				   addr);
 		if (e)
 			break;
